modules/ui/src/test/scala/lucuma/ui/optics/TruncatedRefinedBigDecimalSpec.scala

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.ui.optics
+
+import cats.kernel.laws.discipline.OrderTests
+import eu.timepit.refined.cats._
+import lucuma.core.optics.laws.discipline.SplitEpiTests
+import lucuma.ui.optics.arb._
+import munit.DisciplineSuite
+import org.scalacheck.Arbitrary._
+
+class TruncatedRefinedBigDecimalSpec extends DisciplineSuite {
+  import ArbTruncatedRefinedBigDecimal._
+
+  checkAll("TruncatedRefinedBigDecimal",
+           OrderTests[TruncatedRefinedBigDecimal[OneToThree, 1]].order
+  )
+
+  checkAll("TruncatedRefinedBigDecimal.unsafeRefinedBigDecimal",
+           SplitEpiTests(TruncatedRefinedBigDecimal.unsafeRefinedBigDecimal[OneToThree, 1]).splitEpi
```

Comment:
Is it unsafe if you can pass the tests?

Reply:
If a specific refinement/decimal combination passes the tests, it is safe. However, a different combination may be unsafe.